### ISSUE TITLE
Balloting History for Area Director Workload page

### ISSUE THREAD

**Issue by rdanyliw** [ietf-tools/datatracker#4575]
### Description

Add a new section in the Area Director Workload page (https://datatracker.ietf.org/doc/ad/) to cover balloting history.

Title = "Balloting Record"
Column 1: AD Name
Column 2: Total number of documents presented to the AD to ballot on
Column 3: Number of documents on which a COMMENT has been filed
Column 4: Number of documents on which a DISCUSS has been filed
Column 5: Number of documents on which an ABSTAIN has been filed
Column 6: Number of document on which a RECUSE has been filed
Column 7: Number of document on which no ballot has been filed

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by rjsparks** [comment]
By documents, you mean drafts. Other ballot types (charter, conflict-review, status-change) would need different treatment.

How do we deal with multiple positions on a document? If an AD files a DISCUSS and later and ABSTAIN on the same document, both counters would go up?

Why is "YES" and "No Objection" not in the list. Perhaps a counter that shows how many YES or NOOBJ with _no_ comment would be useful?

Note that "COMMENT" isn't a position. We can build counters for drafts with BallotPositionDocEvents that have non-empty comment fields, but think about what this is really going to tell you - I'm not sure that perhaps a count of ballots with no discuss and no comment text might be more useful?

**Comment by rdanyliw** [comment]
Yes, I meant drafts only.

I confused a number of things.  The states I was interested in tracking were what you said -- Yes, NoObj, Discuss, Abstain,  Recuse, and No Record.

As a first pass since it wouldn't entail more complicated logic, each position type would be counted.  For the situation of "If an AD files a DISCUSS and later and ABSTAIN on the same document, both counters would go up?"+1 on DISCUSS and +1 on ABSTAIN.

To the line of inquiry around "no position", I meant a count of a "no record" ballot.

**Comment by evyncke** [comment]
Even if this is not always really meaningful, should the "size" (in lines) of the ballot explanation be counted ? I.e., a very detailed review is more useful than a simple 'empty' YES or NoObj

**Comment by larseggert** [comment]
@evyncke I don't think so. We don't want to encourage verbosity to drive up some score.

**Comment by evyncke** [comment]
@larseggert  OTOH, balloting without any real comments is also driving up scores ;)

**Comment by rjsparks** [comment]
Some notes:

The simplest implementation would count all three of an AD taking a "Yes" then a "Discuss" then back to a "Yes". Finding the _final_ position for a given AD on a given ballot is possible but would involve enough to warrant pre-computation and caching rather than calculating this in-path of a request.

Some documents have more than one approve ballot:
```
>>> for bde in BallotDocEvent.objects.filter(doc__name="draft-ietf-calext-jscontact",type="created_ballot"):
...     print(Counter(BallotPositionDocEvent.objects.filter(ballot=bde,balloter__name="Roman Danyliw").values_list("pos__slug",flat=True)))
... 
Counter({'yes': 2})
Counter({'discuss': 2, 'yes': 1, 'noobj': 1})
```
Calculating "never took a position" for a given AD will have edges, as "no position" is _assumed_ until a position is taken (there's no initial BallotPositionDocEvent). This runs into the "datatracker doesn't know when an AD started" problem, but it _might_ be possible to work around it by first finding the full set of ballots for which any position exists by anyone on the IESG _now_ (the moment the stats are computed), then looping over each ballot and building counts by enumerating which ADs were expected to take a position on that ballot (see BallotDocEvent.all_positions()). (Edit: no - this won't work - the datatracker can only answer who is expected to ballot _now_, not who was expected to ballot at some previous point in time).
